src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Test.Common;
+using System.Runtime.InteropServices;
+
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class SocketOptionNameTest
+    {
+        private static bool IsWindows10Platform
+        {
+            get
+            {
+                return Capability.IsWindows10Platform();
+            }
+        }
+
+        private static bool IsNotWindows10Platform
+        {
+            get
+            {
+                return !Capability.IsWindows10Platform();
+            }
+        }
+
+        [ConditionalFact("IsNotWindows10Platform")]
+        public void ReuseUnicastPort_CreateSocketGetOption_NotWindows10Platform_Throws()
+        {
+            var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
In general we've tried to match exceptions across platforms, even if the feature is otherwise platform unsupported.  For example, there are APIs elsewhere in corefx that will perform argument validation and throw an ArgumentException for an invalid argument, but then once you get past argument validation you get a PlatformNotSupportedException.  I think here it makes sense to behave similarly: since SocketException is what's used on Windows here to denote invalid arguments, you get a SocketException for an invalid argument on Unix as well.